lib/fizzy/execute.cpp

```diff
@@ -189,8 +190,10 @@ Instance instantiate(const Module& module)
 
 execution_result execute(Instance& instance, FuncIdx function, std::vector<uint64_t> args)
 {
-    // TODO: handle the case when function index points to import
-    const auto& code = instance.module.codesec[function];
+    if (function < instance.imported_functions.size())
+        return instance.imported_functions[function](instance, std::move(args));
+
+    const auto& code = instance.module.codesec[function - instance.imported_functions.size()];
```

Comment:
Shouldn't we also add an assert that this value is less than `codesec.size()` ?

Reply:
I did in later commit

Reply:
Yes that is confusing some of that is moved around in later commits. Would it be possible to clean that up?

Reply:
I can squash first three commits